### PR TITLE
Fixed convergence of training

### LINKED ISSUE #695: Convergence of training is broken
### What happened?

Currently, training does not converge as expected

### What are the steps to reproduce the bug?

This happens across all configurations in current use.

### Version

develop

### Platform (OS and architecture)

ALL

### Relevant log output

```shell

```

### Accompanying data

_No response_

### Organisation

_No response_

### PR DESCRIPTION
**Pull request by clessig**
Restored old prediction had functionally. Other adjustments/reverts, in particular in attention.

## Description

In the plots below, the improvements compared to the older runs might be due to changes in the loss computation. Overall, the important point is that training does not stagnate early.

**ERA5 only:**
old: naoj54ch and precursors 
fixed: e0xdl678g 
broken: IIcmk8th

bafgq0yu_av4qh6pm_naoj54ch_e6x9ripj_llcmk8th_ef9hl74a_ie3g4yfp_c8fsgz42_e0xdl78g_val_ERA5

**ERA5 + CERRA:**
old: txwqvm83 and precursors
fixed: i6ezkO
broken: cl46szaf / x2y9ktqe and precursors

bozdnseb_evq2wki6_tn0lkmbz_yt9i2mlp_rz3l5tse_rcsxyrz8_txwqvm83_tozg0we5_i6eczfk0_x2y9ktqe_cl46szqf_fg581pox_val_CERRA

bozdnseb_evq2wki6_tn0lkmbz_yt9i2mlp_rz3l5tse_rcsxyrz8_txwqvm83_tozg0we5_i6eczfk0_x2y9ktqe_cl46szqf_fg581pox_val_ERA5

**ERA5 + SEVIRI:**
old: fcaplwhi / owfhi8nm 
fixed: gwrvips3 / yi3jxhuw

i3q0jytf_kxgvdam2_ssucopnf_f5n4b7kx_seqvotd1_fcaplwhi_owfhi8nm_k4o2ked7_gwrvips3_yi3jxhuw_val_ERA5

i3q0jytf_kxgvdam2_ssucopnf_f5n4b7kx_seqvotd1_fcaplwhi_owfhi8nm_k4o2ked7_gwrvips3_yi3jxhuw_train_SEVIRI

## Type of Change

-   [X] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Issue Number

Closes #695 

## Code Compatibility

-   [X] I have performed a self-review of my code

### Code Performance and Testing

-   [X] I ran the `uv run train` and (if necessary) `uv run evaluate` on a least one GPU node and it works 
-   [ ] If the new feature introduces modifications at the config level, I have made sure to have notified the other software developers through Mattermost and updated the paths in the `$WEATHER_GENERATOR_PRIVATE` directory

### Dependencies

-   [ ] I have ensured that the code is still pip-installable after the changes and runs
-   [ ] I have tested that new dependencies themselves are pip-installable.
-   [ ] I have not introduced new dependencies in the inference portion of the pipeline

### Documentation

-   [ ] My code follows the style guidelines of this project
-   [ ] I have updated the documentation and docstrings to reflect the changes
-   [ ] I have added comments to my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/weathergen/model/attention.py [hunk 1]
```diff
@@ -172,16 +170,14 @@ def forward(self, x, x_lens=None):
 
         outs = self.compiled_flex_attention(qs, ks, vs).transpose(1, 2).squeeze()
 
-        x = self.proj_out(outs.flatten(-2, -1))
+        out = self.dropout(self.proj_out(outs.flatten(-2, -1)))
```
Comment (sophie-xhonneux) [inline src/weathergen/model/attention.py:172]: This additional dropout is odd to me, because flash attention already applies dropout, but maybe this extra dropout is important somehow?

File: src/weathergen/model/attention.py [hunk 2]
```diff
@@ -461,33 +450,31 @@ def forward(self, x_q, x_kv, x_q_lens=None, x_kv_lens=None, ada_ln_aux=None):
...
-                    x_q_lens.max().item(),
-                    x_kv_lens.max().item(),
+                    x_q_lens.max(),
+                    x_kv_lens.max(),
                     softcap=self.softcap,
                     dropout_p=self.dropout_rate,
                 )
             ]
 
+        # outs = self.dropout( self.proj_out( torch.stack(outs).transpose(1,0).flatten( -2, -1)) )
```
Comment (sophie-xhonneux) [inline src/weathergen/model/attention.py]: maybe best to clean up the commented out line? Also here we make the opposite choice of not having double dropout: how come?

File: src/weathergen/model/attention.py [hunk 3]
```diff
@@ -499,9 +486,7 @@ def __init__(
         self.with_flash = with_flash
         self.dropout_rate = dropout_rate
         self.with_residual = with_residual
-        self.softcap = softcap
```
Comment (sophie-xhonneux) [inline src/weathergen/model/attention.py:502]: The changes to this class, I don't quite follow. It still uses flash attention, but you cannot set the softcap value anymore?

File: src/weathergen/model/blocks.py [hunk 4]
```diff
@@ -199,35 +200,53 @@ def __init__(
...
         self.block.append(
             MultiCrossAttentionHeadVarlen(
                 dim_in,
-                dim_kv,
-                num_heads,
+                self.cf.ae_global_dim_embed,
+                self.cf.streams[0]["target_readout"]["num_heads"],
                 dim_head_proj=self.tr_dim_head_proj,
                 with_residual=True,
-                **attention_kwargs,
```
Comment (sophie-xhonneux) [inline src/weathergen/model/blocks.py:210]: what is the problem with using the attention_kwargs, which makes the code a lot more readable imho

Comment (clessig) [reply src/weathergen/model/blocks.py:210]: No known problem. It was just easier since the old code didn't used the `kwargs`. We can add it back in in a separate PR.

File: src/weathergen/model/attention.py (hunk 2 above)
Comment (clessig) [reply src/weathergen/model/attention.py]: Done.

File: src/weathergen/model/attention.py (hunk 3 above)
Comment (clessig) [reply src/weathergen/model/attention.py:502]: Fixed.

File: src/weathergen/model/attention.py (hunk 1 above)
Comment (clessig) [reply src/weathergen/model/attention.py:172]: This is torch's flex attention. It doesn't have the dropout built in.
